### PR TITLE
Deprecate old JSON reader (#3610)

### PR DESCRIPTION
**Pull request by tustvold**
# Which issue does this PR close?

Part of #3610

# Rationale for this change
 

Deprecate the old reader to encourage people to switch to the new reader

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: parquet/src/arrow/arrow_writer/levels.rs [hunk 1]
```diff
@@ -1121,7 +1121,7 @@ mod tests {
         // Note: we are using the JSON Arrow reader for brevity
         let json_content = r#"
         {"stocks":{"long": "$AAA", "short": "$BBB"}}
-        {"stocks":{"long": null, "long": "$CCC", "short": null}}
+        {"stocks":{"long": "$CCC", "short": null}}
```
Comment (tustvold) [inline parquet/src/arrow/arrow_writer/levels.rs:1124]: This wasn't actually valid JSON, it contained a duplicate key - the RawReader handles this differently than the old JSON reader

File: arrow-json/src/reader.rs [hunk 2]
```diff
@@ -582,6 +582,7 @@ where
 /// [`RawDecoder`]: crate::raw::RawDecoder
 /// [#3610]: https://github.com/apache/arrow-rs/issues/3610
 #[derive(Debug)]
+#[deprecated(note = "Use RawDecoder instead")]
```
Comment (viirya) [inline arrow-json/src/reader.rs:585]: Still need the comment? Actually it reads unclear and seems strange to put issue link there.

```suggestion
#[derive(Debug)]
#[deprecated(note = "Use RawDecoder instead")]
```

File: arrow-json/src/reader.rs [hunk 3]
```diff
@@ -1604,12 +1606,15 @@ fn flatten_json_string_values(values: &[Value]) -> Vec<Option<String>> {
 /// [`RawReader`]: crate::raw::RawReader
 /// [#3610]: https://github.com/apache/arrow-rs/issues/3610
```
Comment (viirya) [inline arrow-json/src/reader.rs:1607]: ditto.

```suggestion
```

**Review by viirya — APPROVED** [review]

File: arrow-json/src/reader.rs (hunk 2 above)
Comment (tustvold) [reply arrow-json/src/reader.rs:585]: These are link shorthand, they don't get rendered, but are used to attach links to the text above.

image

File: parquet/src/arrow/arrow_writer/levels.rs (hunk 1 above)
Comment (tustvold) [reply parquet/src/arrow/arrow_writer/levels.rs:1124]: The prior behaviour wasn't documented on `Decoder`, so I think this is fine

**Pull request merged**